src/ngx_http_lua_common.h

```diff
@@ -147,6 +149,13 @@ typedef struct {
 } ngx_http_lua_preload_hook_t;
 
 
+typedef struct {
+    ssize_t              max_size;
+    size_t               size;
+    u_char              *data;
```

Comment:
Instead of a flat static buffer, I think we should use a ring buffer that can

1. distinguish between error log message boundaries so that we don't have to split it later ourselves (some error log messages do span multiple lines, like those with a Lua backtrace, which makes late splitting nontrivial), and
2. when the buffer is full, the oldest log messages are automatically discarded and overwritten. A ring buffer design can avoid moving existing log data inside the buffer for this feature.